graph/graph.go

```diff
@@ -29,9 +29,10 @@ import (
 
 // A Graph is a store for versioned filesystem images and the relationship between them.
 type Graph struct {
-	Root    string
-	idIndex *truncindex.TruncIndex
-	driver  graphdriver.Driver
+	Root     string
+	idIndex  *truncindex.TruncIndex
+	driver   graphdriver.Driver
+	retained map[string][]string
```

Comment:
It might be helpful to make this a type. I also think we'll need some locking for it.